astropy/nddata/nddata.py

```diff
@@ -37,20 +30,15 @@ class NDData(NDDataBase):
     Parameters
     -----------
     data : `~numpy.ndarray`, `~numpy.ndarray`-like, or `NDData`
-        The actual data contained in this `NDData` object. If possible, data
-        will not be copied`data`, so you should make copy the ``data`` before
-        passing it in if that's the desired behavior.
+        The actual data contained in this `NDData` object.
 
     uncertainty : any type, optional
-        Uncertainty on the data. The uncertainty *must* have a string attribute
-        named ``uncertainty_type``, but there is otherwise no restriction.
+        Uncertainty on the data. The uncertainty *should* have a string
```

Comment:
@mwcraig I just realized that there is already a PR which implements an ``UnknownUncertainty`` class (#3714) if you really do not like the *should* I could implement an adapted version of ``UnknownUncertainty``. I could alter the ``getter`` so that anyone using an unknown uncertainty is 'believing' ``NDData`` has saved their ``value`` but in truth it is something that subclasses from ``NDUncertainty`` with the corresponding properties and methods (but altered in a way that it cannot be propagated with anything not even with itself). That would actually be only be a minor change.